app/main/log-utils.js

```diff
@@ -0,0 +1,86 @@
+/*
+ Copyright 2016 Mozilla
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the
+ License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed
+ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ specific language governing permissions and limitations under the License.
+ */
+
+import path from 'path';
+import fs from 'fs-promise';
+
+/**
+ * This file is loaded in app/shared/logging via `tofino-log-utils` alias,
+ * since we don't want to include node APIs in the renderer process.
+ */
+
+const PATHS = {
+  linux: env => [
+    [env['XDG_CONFIG_HOME']],
+    [env['HOME'], '.config'],
+    [env['XDG_DATA_HOME']],
+    [env['HOME'], '.local', '.share'],
```

Comment:
Do we really want to allow so much variability here? Why not just log everything to a file in the profile directory? Just import environment.js. See how session.json is created.